staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -140,6 +145,70 @@ func (record *attributesRecord) AddAnnotation(key, value string) error {
 	return nil
 }
 
+func (record *attributesRecord) GetReinvocationContext() ReinvocationContext {
```

Comment:
this makes a shallow copy. You call mutating methods on this further down and therefore miss those changes.

Reply:
`record.reinvocationContext` is assigned to a pointer of `reinvocationContext`.

Reply:
it's an interface. Nvmd.